docs/source/TestParameters.rst

```diff
@@ -0,0 +1,945 @@
+   Variant 3:    /os/distro/redhat/fedora/version/20, /os/distro/redhat/fedora/flavor/cloud, /os/arch/i386
+   Variant 4:    /os/distro/redhat/fedora/version/20, /os/distro/redhat/fedora/flavor/cloud, /os/arch/x86_64
+   Variant 5:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/workstation, /os/arch/i386
+   Variant 6:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/workstation, /os/arch/x86_64
+   Variant 7:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/cloud, /os/arch/i386
+   Variant 8:    /os/distro/redhat/fedora/version/21, /os/distro/redhat/fedora/flavor/cloud, /os/arch/x86_64
+   Variant 9:    /os/distro/redhat/rhel/5, /os/arch/i386
+   Variant 10:    /os/distro/redhat/rhel/5, /os/arch/x86_64
+   Variant 11:    /os/distro/redhat/rhel/6, /os/arch/i386
+   Variant 12:    /os/distro/redhat/rhel/6, /os/arch/x86_64
+
+Versus Jenkin's sparse matrix::
+
+   os_version = fedora20 fedora21 rhel5 rhel6
+   os_flavor = none workstation cloud
+   arch = i386 x86_64
+
+   filter = ((os_version == "rhel5").implies(os_flavor == "none") &&
+             (os_version == "rhel6").implies(os_flavor == "none")) &&
+            !(os_version == "fedora20" && os_flavor == "none") &&
+            !(os_version == "fedora21" && os_flavor == "none")
+
+Which is still relatively simple example, but it grows dramatically with
+inner-dependencies.
+
+MuxPlugin
+~~~~~~~~~
+
+:mod:`avocado.core.mux.MuxPlugin`
+
+Defines the full interface required by
+:mod:`avocado.core.plugin_interfaces.VarianterPlugin`. The plugin writer
+should inherit from this ``MuxPlugin``, then from the ``VarianterPlugin``
+and call the::
+
+   self.initialize_mux(root, mux_path, debug)
+
+Where:
+
+* root - is the root of your params tree (compound of `TreeNode`_ -like
+  nodes)
+* mux_path - is the `Mux path`_ to be used in test with all variants
+* debug - whether to use debug mode (requires the passed tree to be
+  compound of ``TreeNodeDebug``-like nodes which stores the origin
+  of the variant/value/environment as the value for listing purposes
+  and is __NOT__ intended for test execution.
+
+This method must be called before the `Varianter`_'s second stage
+(the latest opportunity is during ``self.update_defaults``). The
+`MuxPlugin`_'s code will take care of the rest.
+
+MuxTree
+~~~~~~~
+
+This is the core feature where the hard work happens. It walks the tree
+and remembers all leaf nodes or uses list of `MuxTrees` when another
+multiplex domain is reached while searching for a leaf.
+
+When it's asked to report variants, it combines one variant of each
+remembered item (leaf node always stays the same, but `MuxTree` circles
+through it's values) which recursively produces all possible variants
+of different `multiplex domains`_.
+
+Yaml_to_mux plugin
+==================
+
+:mod:`avocado.plugins.yaml_to_mux`
```

Comment:
Actually I left it there as I use the same on many places, but I also updated the `yaml_to_mux` to point to the same location.